Services/Simulation/JavascriptInterpreter.cs

```diff
@@ -84,16 +93,21 @@ public class JavascriptInterpreter : IJavascriptInterpreter
                 }
 
                 this.log.Debug("Executing JS function", () => new { filename });
-                JsValue output = engine.Execute(program).Invoke("main", context, this.deviceState);
 
-                var result = this.JsValueToDictionary(output);
-                this.log.Debug("JS function success", () => new { filename, result });
-                return result;
+                JsValue output = engine.Execute(program).Invoke(
+                    "main",
+                    context,
+                    this.deviceState.GetAll(),
```

Comment:
I'm assuming state & properties are passed byval - will the script potentially update state & properties directly (other than throught he callbacks)?

Reply:
That's the intention behind the wrapper for the concurrent dictionary, which should only allow for updates to the actual state via a state.Set()